src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1433,24 +1501,24 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 1);
             assert(simdBaseType == TYP_FLOAT);
-
-            switch (simdSize)
+            if (simdSize == 64 && IsBaselineVector512IsaSupportedOpportunistically())
             {
-                case 16:
-                    intrinsic = NI_SSE2_ConvertToVector128Int32WithTruncation;
-                    break;
-                case 32:
-                    intrinsic = NI_AVX_ConvertToVector256Int32WithTruncation;
-                    break;
-                case 64:
-                    intrinsic = NI_AVX512F_ConvertToVector512Int32WithTruncation;
-                    break;
-                default:
-                    unreached();
+                op1       = impSIMDPopStack();
+                intrinsic = NI_AVX512F_ConvertToVector512Int32WithTruncation;
+                retNode   = gtNewSimdCvtNode(retType, op1, intrinsic, CORINFO_TYPE_INT, simdBaseJitType, simdSize);
+            }
+            else if (simdSize == 32 && compOpportunisticallyDependsOn(InstructionSet_AVX))
+            {
+                op1       = impSIMDPopStack();
+                intrinsic = NI_AVX_ConvertToVector256Int32WithTruncation;
+                retNode   = gtNewSimdCvtNode(retType, op1, intrinsic, CORINFO_TYPE_INT, simdBaseJitType, simdSize);
+            }
+            else if (simdSize == 16 && compOpportunisticallyDependsOn(InstructionSet_SSE41))
+            {
+                op1       = impSIMDPopStack();
+                intrinsic = NI_SSE2_ConvertToVector128Int32WithTruncation;
+                retNode   = gtNewSimdCvtNode(retType, op1, intrinsic, CORINFO_TYPE_INT, simdBaseJitType, simdSize);
```

Comment:
It's not clear why this had to change from the switch?

Couldn't we have just changed `retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);` to `retNode = gtNewSimdCvtNode(retType, op1, intrinsic, CORINFO_TYPE_INT, simdBaseJitType, simdSize);`

It's also not clear why `simdSize == 16` is limited on `SSE41`. The actual `gtNewSimdCvtNode` path only uses `AND`, `EQ`, `GE`, and `CndSel`, all of which have paths supporting back to SSE2.